test/common/network/address_impl_test.cc

```diff
@@ -100,8 +100,10 @@ TEST(Ipv4InstanceTest, BadAddress) {
 
 TEST(Ipv4InstanceTest, SocketBindAndConnect) {
   // Test listening on and connecting to an unused port on the IPv4 loopback address.
-  Ipv4Instance loopbackPort("127.0.0.1", ::Network::Test::getUnusedPort());
-  testSocketBindAndConnect(loopbackPort);
+  InstancePtr addrPort(new Ipv4Instance("127.0.0.1", 0));
+  addrPort = Network::Test::checkPortAvailability(addrPort, SocketType::Stream);
```

Comment:
As you mentioned later in your review, this is racy because it doesn't provide a means to reserve the port against another thread/process allocating the same port. For that reason, I don't think "allocate" makes a good prefix.

I will add a variant that takes a string.